src/Generator/CreditMemoGenerator.php

```diff
@@ -117,7 +123,7 @@ private function getToAddress(?ChannelShopBillingData $channelShopBillingData):
             return null;
         }
 
-        return new ShopBillingData(
+        return $this->shopBillingDataFactory->createWithData(
```

Comment:
Maybe a method in factory called `createWithChannelShopBillingData` ? 😉 
See: https://github.com/Sylius/RefundPlugin/pull/313/files#diff-dc7a9ca1a71645aa3eb35cf2d342ab0543241c3b0173c6fe7d6ddf42a641be35R65

It won't block a merge imo - just a suggestion.

Reply:
i would leave it as it is now